packages/Kokkos/CMakeLists.txt

```diff
@@ -1,6 +1,58 @@
+INCLUDE(CombinedOption)
+INCLUDE(TribitsETISupport)
+
 TRIBITS_SUBPACKAGE(Kokkos)
 
-#ADD_SUBDIRECTORY(src)
+TRIBITS_ADD_EXPLICIT_INSTANTIATION_OPTION()
+
+TRIBITS_ADD_ETI_SUPPORT()
+
+ASSERT_DEFINED(${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION)
+
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  DataTransferKit_ENABLE_EXPLICIT_INSTANTIATION
+  HAVE_DATATRANSFERKIT_EXPLICIT_INSTANTIATION
+  "Enable explicit template instantiation (ETI) in DTK"
+  ${${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION}
+  )
```

Comment:
I added this at the package level to mimic the Tpetra/TpetraCore ETI configuration. It seemed like a good idea to separate them like this too while we are refactoring in this directory. I suspect when we are done refactoring we will be able to unify everything and may even have a single source directory.